dds/DCPS/TypeAssignability.cpp

```diff
@@ -0,0 +1,2083 @@
+      }
+
+      switch (ta.minimal.kind) {
+      case TK_ANNOTATION:
+        return assignable_annotation(ta.minimal, tb.minimal);
+      case TK_STRUCTURE:
+        return assignable_struct(ta.minimal, tb.minimal);
+      case TK_UNION:
+        return assignable_union(ta.minimal, tb.minimal);
+      case TK_BITSET:
+        return assignable_bitset(ta.minimal, tb.minimal);
+      case TK_SEQUENCE:
+        return assignable_sequence(ta.minimal, tb.minimal);
+      case TK_ARRAY:
+        return assignable_sequence(ta.minimal, tb.minimal);
+      case TK_MAP:
+        return assignable_map(ta.minimal, tb.minimal);
+      case TK_ENUM:
+        return assignable_enum(ta.minimal, tb.minimal);
+      case TK_BITMASK:
+        return assignable_bitmask(ta.minimal, tb.minimal);
+      default:
+        return assignable_extended(ta.minimal, tb.minimal);
+      }
+    }
+
+    return false;
+  }
+
+  /**
+   * @brief Both input can be of any type
+   */
+  bool TypeAssignability::assignable(const TypeIdentifier& ta,
+                                     const TypeIdentifier& tb) const
+  {
+    switch (ta.kind) {
+    case TK_BOOLEAN:
+    case TK_BYTE:
+    case TK_INT16:
+    case TK_INT32:
+    case TK_INT64:
+    case TK_UINT16:
+    case TK_UINT32:
+    case TK_UINT64:
+    case TK_FLOAT32:
+    case TK_FLOAT64:
+    case TK_FLOAT128:
+    case TK_INT8:
+    case TK_UINT8:
+    case TK_CHAR8:
+    case TK_CHAR16:
+      return assignable_primitive(ta, tb);
+    case TI_STRING8_SMALL:
+    case TI_STRING8_LARGE:
+    case TI_STRING16_SMALL:
+    case TI_STRING16_LARGE:
+      return assignable_string(ta, tb);
+    case TI_PLAIN_SEQUENCE_SMALL:
+    case TI_PLAIN_SEQUENCE_LARGE:
+      return assignable_plain_sequence(ta, tb);
+    case TI_PLAIN_ARRAY_SMALL:
+    case TI_PLAIN_ARRAY_LARGE:
+      return assignable_plain_array(ta, tb);
+    case TI_PLAIN_MAP_SMALL:
+    case TI_PLAIN_MAP_LARGE:
+      return assignable_plain_map(ta, tb);
+    case TI_STRONGLY_CONNECTED_COMPONENT:
+      // No rule in the spec for strongly connected components
+      return false;
+    case EK_COMPLETE:
+      // Assuming only equivalence kind of EK_MINIMAL is supported
+      return false;
+    case EK_MINIMAL: {
+      const MinimalTypeObject& base_type_a = typelookup_.lookup_minimal(ta);
+      const MinimalTypeObject& base_type_b = typelookup_.lookup_minimal(tb);
+      TypeObject wrapper_a(base_type_a), wrapper_b(base_type_b);
+      return assignable(wrapper_a, wrapper_b);
+    }
+    default:
+      return false; // Future extensions
+    }
+  }
+
+  /**
+   * @brief At least one input type object must be TK_ALIAS
+   */
+  bool TypeAssignability::assignable_alias(const MinimalTypeObject& ta,
+                                           const MinimalTypeObject& tb) const
+  {
+    if (TK_ALIAS == ta.kind && TK_ALIAS != tb.kind) {
+      const TypeIdentifier& tia = *ta.alias_type.body.common.related_type.in();
```

Comment:
```suggestion
      const TypeIdentifier& tia = *ta.alias_type.body.common.related_type;
```
I don't think `.in()` would be needed.  Of course, keep it in if required to compile.

Reply:
Removing `.in()` compiles, so I've removed it.